AI/VCAI/ResourceManager.cpp

```diff
@@ -0,0 +1,304 @@
+/*
+* ResourceManager.cpp, part of VCMI engine
+*
+* Authors: listed in file AUTHORS in main folder
+*
+* License: GNU General Public License v2.0 or later
+* Full text of license available in license.txt file, in main folder
+*
+*/
+#include "StdInc.h"
+#include "ResourceManager.h"
+
+#include "../../CCallback.h"
+#include "../../lib/mapObjects/MapObjects.h"
+
+#define GOLD_RESERVE (10000); //at least we'll be able to reach capitol
+
+ResourceManager * rm;
+
+ResourceObjective::ResourceObjective(TResources & Res, Goals::TSubgoal Goal)
+	: resources(Res), goal(Goal)
+{
+}
+
+//bool operator<(const ResourceObjective & lhs, const ResourceObjective & rhs)
+//{
+//	return lhs.goal->priority < rhs.goal->priority;
+//}
+//bool operator<(const ResourceObjective lhs, const ResourceObjective rhs)
+//{
+//	return lhs.goal->priority < rhs.goal->priority;
+//}
+
+bool ResourceObjective::operator<(const ResourceObjective & ro) const
+{
+	return goal->priority < ro.goal->priority;
+}
+
+ResourceManager::ResourceManager(CPlayerSpecificInfoCallback * CB, VCAI * AI)
+	: ai(AI), cb(CB)
+{
+}
+
+void ResourceManager::setCB(CPlayerSpecificInfoCallback * CB)
+{
+	cb = CB;
+}
+
+void ResourceManager::setAI(VCAI * AI)
+{
+	ai = AI;
+}
+
+bool ResourceManager::canAfford(const TResources & cost) const
+{
+	return freeResources().canAfford(cost);
+}
+
+TResources ResourceManager::estimateIncome() const
```

Comment:
I do get this might be beyond what you want to touch during this refactoring, but it's could be great to create unified income calculation code within libvcmi since current one is a mess: code that handle thieves guild is separate and one that handle actual income is separate as well.